src/ToricVarieties/cohomCalg/ShiftedCones/attributes.jl

```diff
@@ -0,0 +1,158 @@
+############################
+# Attributes
+############################
+
+@doc Markdown.doc"""
+    gens(sc::ShiftedCone)
+
+Return the generators of the shifted cone `sc`.
+
+# Examples
+```jldoctest
+julia> ray_generators = [[1,2,3],[4,5,6]]
+2-element Vector{Vector{Int64}}:
+ [1, 2, 3]
+ [4, 5, 6]
+
+julia> shift = [-3,-2,-1]
+3-element Vector{Int64}:
+ -3
+ -2
+ -1
+
+julia> sc = ShiftedCone(ray_generators, shift)
+A shifted cone in ambient dimension 3
+
+julia> gens(sc)
+[1   2   3]
+[4   5   6]
+```
+"""
+function gens(sc::ShiftedCone)
+    return sc.gens::fmpz_mat
+end
+export gens
+
+
+@doc Markdown.doc"""
+    ngens(sc::ShiftedCone)
+
+Return the number of generators of the shifted cone `sc`.
+
+# Examples
+```jldoctest
+julia> ray_generators = [[1,2,3],[4,5,6]]
+2-element Vector{Vector{Int64}}:
+ [1, 2, 3]
+ [4, 5, 6]
+
+julia> shift = [-3,-2,-1]
+3-element Vector{Int64}:
+ -3
+ -2
+ -1
+
+julia> sc = ShiftedCone(ray_generators, shift)
+A shifted cone in ambient dimension 3
+
+julia> ngens(sc)
+2
+```
+"""
+function ngens(sc::ShiftedCone)
+    return fmpz(nrows(sc.gens))::fmpz
```

Comment:
Since `sc.gens` is knowns to be of type `fmpz_mat`, also the type of `nrows(sc.gens)` is known, and hence the type assertion is not needed.

In addition, though, I wonder whether it makes sense to return an `fmpz` here instead of an `Int`? Surely you don't plan to support more than ~2^63 generators (and in any case, `fmpz_mat` doesn't support this)...

In other words, why not just this?
```suggestion
    return nrows(sc.gens)
```

Reply:
Long story short: you are right. It seems I was carried away with my use of `fmpz`.